### PR TITLE
ISPN-5495 ConcurrentStartTest.testConcurrentStart random failures

### PR DESCRIPTION
**Pull request by danberindei**
https://issues.jboss.org/browse/ISPN-5495
https://issues.jboss.org/browse/ISPN-6233
https://issues.jboss.org/browse/ISPN-6235

### CODE REVIEW

File: core/src/main/java/org/infinispan/topology/ClusterTopologyManagerImpl.java [hunk 1]
```diff
@@ -449,10 +449,17 @@ private void confirmMembersAvailable() throws Exception {
...
-   private void waitForView(int joinerViewId, long timeout) throws InterruptedException {
-      if (joinerViewId > viewId) {
-         log.tracef("Received a cache topology command with a higher view id: %s, our view id is %s",
-               joinerViewId, viewId);
+   private void waitForJoinerView(Address joiner, int joinerViewId, long timeout)
+         throws InterruptedException {
+      if (joinerViewId > viewId || clusterManagerStatus == ClusterManagerStatus.RECOVERING_CLUSTER) {
+         if (trace) {
+            if (joinerViewId > viewId) {
+               log.tracef("Waiting to install view %s before processing join request from %s", joiner,
```
Comment (pruivo) [inline core/src/main/java/org/infinispan/topology/ClusterTopologyManagerImpl.java]: tracef parameters in the wrong order (first if the view, then the joiner address)

**Comment by pruivo** [comment]
for some reason, the core test-suite runs in ~25min. Before it was ~7min (3 times worse). Does it happen to you too?

**Comment by danberindei** [comment]
Updated! Great to hear it was just a fluke @pruivo, you had me scared for a bit!

**Comment by pruivo** [comment]
ConcurrentStartTest is still failing:
https://gist.github.com/pruivo/cc4f2027968225d2becb

**Comment by gustavonalle** [comment]
@danberindei Is this sorted out?

**Comment by danberindei** [comment]
@gustavonalle I missed Pedro's comment, trying to reproduce.

**Comment by pruivo** [comment]
integrated! thanks @danberindei !

**Pull request closed without merging**